iocore/net/QUICNetVConnection.cc

```diff
@@ -79,6 +79,26 @@ QUICNetVConnection::init(QUICConnectionId original_cid, UDPConnection *udp_con,
                static_cast<uint64_t>(this->_quic_connection_id));
 }
 
+void
+QUICNetVConnection::do_io_close(int lerrno)
+{
+  QUICConDebug("close connection");
+  if (!this->is_closed()) {
+    // FIXME: should call do_io_read and do_io_write;
+    this->read.enabled       = 1;
+    this->read.vio._cont     = this;
+    this->read.vio.mutex     = this->mutex;
+    this->read.vio.nbytes    = INT64_MAX;
+    this->read.vio.ndone     = 0;
+    this->read.vio.vc_server = this;
+
+    this->_unschedule_packet_write_ready();
+
+    this->_switch_to_closing_state(QUICConnectionErrorUPtr(new QUICConnectionError(QUICTransErrorCode::NO_ERROR)));
```

Comment:
We should probably reduce a path for switching state. Can we call `QUICNetVConnectionclose()` here instead?

Reply:
Do you means we should use close instead of _switch_to_closing_state ?

Reply:
Yes, if it was reasonable. The spec defines three kind of closing, but it's not clear what kind of close we are initiating here. My point is that we should have only one function that initiates "immediate close". So I'm also fine if a same function that initiates "immediate close" is called from here and `close()`.